compiler/resolution/functionResolution.cpp

```diff
@@ -6722,7 +6722,6 @@ static void resolveExprMaybeIssueError(CallExpr* call) {
       if (FnSymbol* fn = callStack.v[head]->resolvedFunction())
         outerCompilerErrorMap[fn] = str;
     } else {
-      gdbShouldBreakHere();
```

Comment:
@mppf: Was my guess correct that this was debugging code that accidentally got committed, and is OK to remove?

Reply:
It's OK to remove it but I'm not sure it was an accident. Nonetheless removing it is probably a good idea (the alternative would be to make it break only if developer). USR_WARN doesn't call gdbShouldBreakHere and we probably don't want it to.

Reply:
My mental pattern for gdbShouldBreakHere() is (and has always been) "after hitting an error, if I re-run the same command with the `--gdb` / `--lldb` flags, the program should run and break at that error before exiting.  But when there are other gdbShouldBreakHere() calls scattered throughout the code, then it stops early, takes me a bit to realize that the code I'm inspecting has nothing to do with the error I was trying to debug, and hit continue to move on to the real error (or maybe the next premature gdbShouldBreakHere().  I can imagine developers adding additional gdbShouldBreakHere() calls while they're working on an issue, but would be surprised that they'd be added back to master, at least enabled by default (I could see guarding them by an #ifdef or something like that).  I don't think guarding them with a developer check would be sufficient to make me happy because I'm almost always going to be debugging with developer on, which would continue to give the current behavior.

What's the counterargument / rationale for checking in this gdbShouldBreakHere() check?  What will make the typical developer want to have the compiler break here?

Reply:
I'm not sure the typical developer wants the compiler to break there but in this case it's useful for debugging warnings coming up during resolution (that weren't expected).

I do add additional gdbShouldBreakHere() calls but that are in something like --break-on-resolve-id (which arguably you meant by guarding by something) but other than that I think your philosophy about them makes sense.